node_modules/oae-content/tests/test-guests.js

```diff
@@ -0,0 +1,160 @@
+            assert.ok(!err);
+
+            // Create a content item
+            RestAPI.Content.createLink(simong.restContext, 'Apereo', 'Apereo', 'private', 'apereo.org', [], [], [], function(err, link) {
+                assert.ok(!err);
+
+                // Make a guest a viewer of the content item through the content permissions API
+                var email = TestsUtil.generateTestEmailAddress();
+                var roleChanges = {};
+                roleChanges[email] = 'viewer';
+                RestAPI.Content.updateMembers(simong.restContext, link.id, roleChanges, function(err) {
+                    assert.ok(!err);
+                });
+                
+                // An email should've been sent to the guest
+                EmailAPI.once('debugSent', function(message) {
+                    // Verify we sent an email to the invited user
+                    assert.strictEqual(message.to[0].address, email);
+
+                    // Verify a token is passed in the email
+                    var token = message.text.match(/\?token=([a-zA-Z0-9]{32})/)[1];
+                    assert.ok(token);
+
+                    // Sign in as the guest
+                    PrincipalsTestUtil.signIntoGuestAccountWithToken(token, function(loggedinGuest) {
+
+                        // The guest should have access to the piece of content
+                        RestAPI.Content.getContent(loggedinGuest.restContext, link.id, function(err, link) {
+                            assert.ok(!err);
+
+                            // Verify the guest is in the members list
+                            RestAPI.Content.getMembers(simong.restContext, link.id, null, null, function(err, members) {
+                                assert.ok(!err);
+                                assert.strictEqual(members.results.length, 2);
+                                var guest = _.findWhere(members.results, {'role': 'viewer'});
+                                assert.ok(guest);
+
+                                // Make the guest a manager
+                                var roleChanges = {};
+                                roleChanges[email] = 'manager';
+                                RestAPI.Content.updateMembers(simong.restContext, link.id, roleChanges, function(err) {
+                                    assert.ok(!err);
+
+                                    // Verify the guest is a manager
+                                    RestAPI.Content.getContent(loggedinGuest.restContext, link.id, function(err, link) {
+                                        assert.ok(!err);
+                                        assert.ok(link.isManager);
+                                        return callback();
+                                    });
+                                });
+                            });
+                        });
+                    });
+                });
+            });
+        });
+    });
+});
```

Comment:
We should also verify that using the email address of an existing user correctly shares the content with that user